include/picongpu/particles/atomicPhysics2/initElectrons/InitElectronAs.hpp

```diff
@@ -0,0 +1,266 @@
+            constexpr float_64 mcElectron = picongpu::SI::ELECTRON_MASS_SI * picongpu::SI::SPEED_OF_LIGHT_SI
+                / (UNIT_MASS * UNIT_LENGTH / UNIT_TIME);
+            // UNIT_MASS * UNIT_LENGTH / UNIT_TIME, not weighted
+            constexpr float_64 mcIon = massIon * picongpu::SI::SPEED_OF_LIGHT_SI / (UNIT_LENGTH / UNIT_TIME);
+
+            // UNIT_MASS * UNIT_LENGTH^2 / UNIT_TIME^2, not weighted
+            constexpr float_64 mSquaredCSquaredIon
+                = pmacc::math::cPow(massIon * picongpu::SI::SPEED_OF_LIGHT_SI / (UNIT_LENGTH / UNIT_TIME), 2u);
+
+            // norm of electron momentum after inelastic collision in Ion-System
+            // weight * (unitless^2 - unitless) * (UNIT_MASS * UNIT_LENGTH/UNIT_TIME)
+            // = UNIT_MASS * UNIT_LENGTH / UNIT_TIME, weighted
+            float_64 const normMomentumStarElectron_IonSystem = static_cast<float_64>(electron[weighting_])
+                * math::sqrt((pmacc::math::cPow(gammaStarElectron_IonSystem, 2u) - 1._X)) * mcElectron;
+
+            //      choose direction
+            float_X const u = rngGenerator();
+            float_X const v = rngGenerator();
+
+            float_X const cosTheta = 1._X - 2._X * v;
+            float_X const sinTheta = math::sqrt(v * (2._X - v));
+            float_X const phi = 2._X * static_cast<float_X>(picongpu::PI) * u;
+
+            float3_64 const directionVector
+                = float3_64(sinTheta * math::cos(phi), sinTheta * math::sin(phi), cosTheta);
+            // UNIT_MASS * UNIT_LENGTH / UNIT_TIME, weighted
+            auto momentumStarElectron_IonSystem = MatrixVector(normMomentumStarElectron_IonSystem * directionVector);
+
+            //      Lorentz transformation from IonSystem to LabSystem
+            // square of original momentum of the ion in Lab-System
+            // UNIT_MASS^2 * UNIT_LENGTH^2 / UNIT_TIME^2, not weighted
+            float_64 const momentumSquaredIon_LabSystem
+                = static_cast<float_64>(pmacc::math::l2norm2(ion[momentum_]) / ion[weighting_]);
+
+            // square of original beta of the ion in Lab-System
+            // beta^2 = 1/(1 + (m^2*c^2)/p^2)
+            //  unitless + (UNIT_MASS^2 * UNIT_SPEED^2)(not weighted)
+            //  /( UNIT_MASS^2 * UNIT_SPEED^2)(not weighted) = unitless
+            float_64 const normSquaredBetaIon_LabSystem
+                = 1._X / (1 + mSquaredCSquaredIon / momentumSquaredIon_LabSystem);
+
+            // unitless
+            float3_64 const betaIonSystem =
+                // magnitude
+                math::sqrt(normSquaredBetaIon_LabSystem)
+                // direction
+                * static_cast<float3_64>(ion[momentum_] / pmacc::math::l2norm(ion[momentum_]));
+
+            // unitless
+            auto beta = MatrixVector(betaIonSystem);
+
+            float_64 const gammaIonSystem
+                = math::sqrt(1. / (1. - pmacc::math::cPow(normSquaredBetaIon_LabSystem, 2u)));
```

Comment:
fixed

Reply:
from a maths and physics point of view an ion beta will always be less than zero, i.e. will never move with light speed.
For very highly relativistic ions floating point rounding errors may create beta values >= 1, but this are highly unlikely, since even a 1GeV hydrogen ion only reaches a beta of 0.7 .
Will create a comment but no check.